052-connector-state.md

```diff
@@ -0,0 +1,55 @@
+<!-- This template is provided as an example with sections you may wish to comment on with respect to your proposal. Add or remove sections as required to best articulate the proposal. -->
+
+# Support stopping connectors
+
+This proposal aims at adding support for stopping Kafka Connect connectors.
+
+## Current situation
+
+At the moment Strimzi allows pausing connectors. This is done by a property, `pause`, in the `KafkaConnectorSpec` schema. Whenever it is set to `true`, the connectors and its tasks are paused.
+
+For example:
+```yaml
+spec:
+  class: org.apache.kafka.connect.file.FileStreamSourceConnector
+  config:
+    file: /opt/kafka/LICENSE
+    topic: my-topic
+  pause: true
+  tasksMax: 1
+```
+
+## Motivation
+
+Since Kafka 3.5, Kafka Connect connectors can also be stopped. This is done by using a new REST endpoint `PUT /connectors/{connector}/stop`. Compared to paused where the connector and tasks stay instantiated, when stopped only the configuration of the connector is kept but nothing is actually running.
+
+A connector paused still uses some resources but it's faster to resume. So using pause is great if you want to halt data flowing for a short duration. For longer durations it's interesting to fully stop the connector with the new API. Also the stopped state will be required to use the offset delete/reset endpoints that should come with Kafka 3.6. For these reasons, Strimzi should support both pausing and stopping connectors.
+
+The `PUT /connectors/{connector}/resume` REST endpoint is used to restart both paused and stopped connectors.
+
+## Proposal
+
+Since a connector can't be paused and stopped at the same time, the proposal is to replace the current `pause` property and add a new property, `state`, in the `AbstractConnectorSpec` schema. That new property will accept the `pause`, `stop` or `run` values.
+
+The `pause` property will be marked as deprecated. In the next Strimzi API version, the `pause` field will be deleted.
+
+This feature requires at least Kafka 3.5 to work. In case it is merged when Strimzi still support older releases, a note in the documentation will be added to notify of this limitation.
```

Comment:
You should specify how the operator would handle `stop` in this case. Should it fail the reconciliation, or ignore the state entirely, or treat `stop` as `pause`? Personally I think treating stopped as paused would be acceptable in this case since that's functionally equivalent.

Reply:
Good point. I took your suggestion and added that when trying to stop a connector with 3.4, it will be paused.